stdlib/public/Backtracing/FileImageSource.swift

```diff
@@ -20,51 +20,54 @@ import Swift
 
 enum FileImageSourceError: Error {
   case posixError(Int32)
-  case truncatedRead
+  case outOfRangeRead
 }
 
 class FileImageSource: ImageSource {
-  private var fd: Int32
+  private var _fd: Int32
+  private var _mapping: UnsafeRawBufferPointer
 
   public var isMappedImage: Bool { return false }
 
   private var _path: String
   public var path: String? { return _path }
 
-  public lazy var bounds: Bounds? = {
-    let size = lseek(fd, 0, SEEK_END)
-    if size < 0 {
-      return nil
-    }
-    return Bounds(base: 0, size: Size(size))
-  }()
+  public var bounds: Bounds? {
+    return Bounds(base: 0, size: Size(_mapping.count))
+  }
 
   public init(path: String) throws {
     _path = path
-    fd = _swift_open(path, O_RDONLY, 0)
-    if fd < 0 {
+    _fd = _swift_open(path, O_RDONLY, 0)
+    if _fd < 0 {
+      throw FileImageSourceError.posixError(_swift_get_errno())
+    }
+    let size = lseek(_fd, 0, SEEK_END)
+    if size < 0 {
       throw FileImageSourceError.posixError(_swift_get_errno())
     }
+    let base = mmap(nil, Int(size), PROT_READ, MAP_FILE|MAP_PRIVATE, _fd, 0)
+    if base == nil || base! == UnsafeRawPointer(bitPattern: -1)! {
+      throw FileImageSourceError.posixError(_swift_get_errno())
```

Comment:
This error case and the one above will leak `_fd`.

Reply:
Good point. Since we don't actually need `_fd` after calling `mmap()` anyway, I think the thing to do is make it a local variable and use a `defer` to clean it up.